From 1f08d9bd69891d24767bd2b2c00ab7f4061b0430 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Mon, 24 Oct 2016 12:40:38 +0100
Subject: [PATCH] Fix Makefile to be compatible with simple autotools based
 autospec run

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 Makefile | 41 +++++++++++++++++++++++++----------------
 lcovrc   |  2 +-
 2 files changed, 26 insertions(+), 17 deletions(-)

diff --git a/Makefile b/Makefile
index 19af0ca..920ada1 100644
--- a/Makefile
+++ b/Makefile
@@ -15,15 +15,20 @@ VERSION := $(shell bin/get_version.sh --version)
 RELEASE := $(shell bin/get_version.sh --release)
 FULL    := $(shell bin/get_version.sh --full)
 
-CFG_DIR := $(PREFIX)/etc
-BIN_DIR := $(PREFIX)/usr/bin
-MAN_DIR := $(PREFIX)/usr/share/man
+PREFIX := /usr
+INSTALL_ROOT := $(PREFIX)/local
+CFG_DIR := $(INSTALL_ROOT)$(PREFIX)/share/defaults/lcov
+BIN_DIR := $(INSTALL_ROOT)$(PREFIX)/bin
+MAN_DIR := $(INSTALL_ROOT)$(PREFIX)/share/man
 TMP_DIR := $(shell mktemp -d)
 FILES   := $(wildcard bin/*) $(wildcard man/*) README Makefile \
 	   $(wildcard rpm/*) lcovrc
 
-.PHONY: all info clean install uninstall rpms
+.PHONY: build all info clean install uninstall rpms
 
+build:
+	@echo "No build for scripts..."
+ 
 all: info
 
 info:
@@ -38,18 +43,22 @@ clean:
 	make -C example clean
 
 install:
-	bin/install.sh bin/lcov $(BIN_DIR)/lcov -m 755
-	bin/install.sh bin/genhtml $(BIN_DIR)/genhtml -m 755
-	bin/install.sh bin/geninfo $(BIN_DIR)/geninfo -m 755
-	bin/install.sh bin/genpng $(BIN_DIR)/genpng -m 755
-	bin/install.sh bin/gendesc $(BIN_DIR)/gendesc -m 755
-	bin/install.sh man/lcov.1 $(MAN_DIR)/man1/lcov.1 -m 644
-	bin/install.sh man/genhtml.1 $(MAN_DIR)/man1/genhtml.1 -m 644
-	bin/install.sh man/geninfo.1 $(MAN_DIR)/man1/geninfo.1 -m 644
-	bin/install.sh man/genpng.1 $(MAN_DIR)/man1/genpng.1 -m 644
-	bin/install.sh man/gendesc.1 $(MAN_DIR)/man1/gendesc.1 -m 644
-	bin/install.sh man/lcovrc.5 $(MAN_DIR)/man5/lcovrc.5 -m 644
-	bin/install.sh lcovrc $(CFG_DIR)/lcovrc -m 644
+	install -d 755 $(BIN_DIR)
+	install -d 755 $(MAN_DIR)/man1
+	install -d 755 $(MAN_DIR)/man5
+	install -d 755 $(CFG_DIR)
+	install -m 755 bin/lcov "$(BIN_DIR)/lcov"
+	install -m 755 bin/genhtml "$(BIN_DIR)/genhtml"
+	install -m 755 bin/geninfo "$(BIN_DIR)/geninfo"
+	install -m 755 bin/genpng "$(BIN_DIR)/genpng"
+	install -m 755 bin/gendesc "$(BIN_DIR)/gendesc"
+	install -m 644 man/lcov.1 "$(MAN_DIR)/man1/lcov.1"
+	install -m 644 man/genhtml.1 "$(MAN_DIR)/man1/genhtml.1"
+	install -m 644 man/geninfo.1 "$(MAN_DIR)/man1/geninfo.1"
+	install -m 644 man/genpng.1 "$(MAN_DIR)/man1/genpng.1"
+	install -m 644 man/gendesc.1 "$(MAN_DIR)/man1/gendesc.1"
+	install -m 644 man/lcovrc.5 "$(MAN_DIR)/man5/lcovrc.5"
+	install -m 644 lcovrc "$(CFG_DIR)/lcovrc"
 	bin/updateversion.pl $(BIN_DIR)/lcov $(VERSION) $(RELEASE) $(FULL)
 	bin/updateversion.pl $(BIN_DIR)/genhtml $(VERSION) $(RELEASE) $(FULL)
 	bin/updateversion.pl $(BIN_DIR)/geninfo $(VERSION) $(RELEASE) $(FULL)
diff --git a/lcovrc b/lcovrc
index 8669bdc..270fd95 100644
--- a/lcovrc
+++ b/lcovrc
@@ -1,5 +1,5 @@
 #
-# /etc/lcovrc - system-wide defaults for LCOV
+# /usr/share/defaults/lcov/lcovrc - system-wide defaults for LCOV
 #
 # To change settings for a single user, place a customized copy of this file
 # at location ~/.lcovrc
-- 
2.10.1

