From bc82dd974867be8fe057ffcaab32945e368a4ebd Mon Sep 17 00:00:00 2001
From: Ikey Doherty <michael.i.doherty@intel.com>
Date: Mon, 15 May 2017 17:01:18 +0100
Subject: [PATCH 1/2] Fix Makefile to be compatible with simple autotools based
 autospec run

Signed-off-by: Ikey Doherty <michael.i.doherty@intel.com>
---
 Makefile | 56 ++++++++++++++++++++++++++++++--------------------------
 lcovrc   |  2 +-
 2 files changed, 31 insertions(+), 27 deletions(-)

diff --git a/Makefile b/Makefile
index 010fec8..f2645aa 100644
--- a/Makefile
+++ b/Makefile
@@ -17,14 +17,18 @@ FULL    := $(shell bin/get_version.sh --full)
 
 PREFIX  := /usr/local
 
-CFG_DIR := $(PREFIX)/etc
+PREFIX := /usr
+CFG_DIR := $(PREFIX)/share/defaults/lcov
 BIN_DIR := $(PREFIX)/bin
 MAN_DIR := $(PREFIX)/share/man
 TMP_DIR := $(shell mktemp -d)
 FILES   := $(wildcard bin/*) $(wildcard man/*) README Makefile \
 	   $(wildcard rpm/*) lcovrc
 
-.PHONY: all info clean install uninstall rpms test
+.PHONY: build all info clean install uninstall rpms test
+
+build:
+	@echo "No build for scripts..."
 
 all: info
 
@@ -40,18 +44,18 @@ clean:
 	make -C example clean
 
 install:
-	bin/install.sh bin/lcov $(DESTDIR)$(BIN_DIR)/lcov -m 755
-	bin/install.sh bin/genhtml $(DESTDIR)$(BIN_DIR)/genhtml -m 755
-	bin/install.sh bin/geninfo $(DESTDIR)$(BIN_DIR)/geninfo -m 755
-	bin/install.sh bin/genpng $(DESTDIR)$(BIN_DIR)/genpng -m 755
-	bin/install.sh bin/gendesc $(DESTDIR)$(BIN_DIR)/gendesc -m 755
-	bin/install.sh man/lcov.1 $(DESTDIR)$(MAN_DIR)/man1/lcov.1 -m 644
-	bin/install.sh man/genhtml.1 $(DESTDIR)$(MAN_DIR)/man1/genhtml.1 -m 644
-	bin/install.sh man/geninfo.1 $(DESTDIR)$(MAN_DIR)/man1/geninfo.1 -m 644
-	bin/install.sh man/genpng.1 $(DESTDIR)$(MAN_DIR)/man1/genpng.1 -m 644
-	bin/install.sh man/gendesc.1 $(DESTDIR)$(MAN_DIR)/man1/gendesc.1 -m 644
-	bin/install.sh man/lcovrc.5 $(DESTDIR)$(MAN_DIR)/man5/lcovrc.5 -m 644
-	bin/install.sh lcovrc $(DESTDIR)$(CFG_DIR)/lcovrc -m 644
+	install -D bin/lcov $(DESTDIR)$(BIN_DIR)/lcov -m 00755
+	install bin/genhtml $(DESTDIR)$(BIN_DIR)/genhtml -m 00755
+	install bin/geninfo $(DESTDIR)$(BIN_DIR)/geninfo -m 00755
+	install bin/genpng $(DESTDIR)$(BIN_DIR)/genpng -m 00755
+	install bin/gendesc $(DESTDIR)$(BIN_DIR)/gendesc -m 00755
+	install -D man/lcov.1 $(DESTDIR)$(MAN_DIR)/man1/lcov.1 -m 00644
+	install man/genhtml.1 $(DESTDIR)$(MAN_DIR)/man1/genhtml.1 -m 00644
+	install man/geninfo.1 $(DESTDIR)$(MAN_DIR)/man1/geninfo.1 -m 00644
+	install man/genpng.1 $(DESTDIR)$(MAN_DIR)/man1/genpng.1 -m 00644
+	install man/gendesc.1 $(DESTDIR)$(MAN_DIR)/man1/gendesc.1 -m 00644
+	install -D man/lcovrc.5 $(DESTDIR)$(MAN_DIR)/man5/lcovrc.5 -m 00644
+	install -D lcovrc $(DESTDIR)$(CFG_DIR)/lcovrc -m 00644
 	bin/updateversion.pl $(DESTDIR)$(BIN_DIR)/lcov $(VERSION) $(RELEASE) $(FULL)
 	bin/updateversion.pl $(DESTDIR)$(BIN_DIR)/genhtml $(VERSION) $(RELEASE) $(FULL)
 	bin/updateversion.pl $(DESTDIR)$(BIN_DIR)/geninfo $(VERSION) $(RELEASE) $(FULL)
@@ -65,18 +69,18 @@ install:
 	bin/updateversion.pl $(DESTDIR)$(MAN_DIR)/man5/lcovrc.5 $(VERSION) $(RELEASE) $(FULL)
 
 uninstall:
-	bin/install.sh --uninstall bin/lcov $(DESTDIR)$(BIN_DIR)/lcov
-	bin/install.sh --uninstall bin/genhtml $(DESTDIR)$(BIN_DIR)/genhtml
-	bin/install.sh --uninstall bin/geninfo $(DESTDIR)$(BIN_DIR)/geninfo
-	bin/install.sh --uninstall bin/genpng $(DESTDIR)$(BIN_DIR)/genpng
-	bin/install.sh --uninstall bin/gendesc $(DESTDIR)$(BIN_DIR)/gendesc
-	bin/install.sh --uninstall man/lcov.1 $(DESTDIR)$(MAN_DIR)/man1/lcov.1
-	bin/install.sh --uninstall man/genhtml.1 $(DESTDIR)$(MAN_DIR)/man1/genhtml.1
-	bin/install.sh --uninstall man/geninfo.1 $(DESTDIR)$(MAN_DIR)/man1/geninfo.1
-	bin/install.sh --uninstall man/genpng.1 $(DESTDIR)$(MAN_DIR)/man1/genpng.1
-	bin/install.sh --uninstall man/gendesc.1 $(DESTDIR)$(MAN_DIR)/man1/gendesc.1
-	bin/install.sh --uninstall man/lcovrc.5 $(DESTDIR)$(MAN_DIR)/man5/lcovrc.5
-	bin/install.sh --uninstall lcovrc $(DESTDIR)$(CFG_DIR)/lcovrc
+	rm -vf bin/lcov $(DESTDIR)$(BIN_DIR)/lcov
+	rm -vf bin/genhtml $(DESTDIR)$(BIN_DIR)/genhtml
+	rm -vf bin/geninfo $(DESTDIR)$(BIN_DIR)/geninfo
+	rm -vf bin/genpng $(DESTDIR)$(BIN_DIR)/genpng
+	rm -vf bin/gendesc $(DESTDIR)$(BIN_DIR)/gendesc
+	rm -vf man/lcov.1 $(DESTDIR)$(MAN_DIR)/man1/lcov.1
+	rm -vf man/genhtml.1 $(DESTDIR)$(MAN_DIR)/man1/genhtml.1
+	rm -vf man/geninfo.1 $(DESTDIR)$(MAN_DIR)/man1/geninfo.1
+	rm -vf man/genpng.1 $(DESTDIR)$(MAN_DIR)/man1/genpng.1
+	rm -vf man/gendesc.1 $(DESTDIR)$(MAN_DIR)/man1/gendesc.1
+	rm -vf man/lcovrc.5 $(DESTDIR)$(MAN_DIR)/man5/lcovrc.5
+	rm -vf lcovrc $(DESTDIR)$(CFG_DIR)/lcovrc
 
 dist: lcov-$(VERSION).tar.gz lcov-$(VERSION)-$(RELEASE).noarch.rpm \
       lcov-$(VERSION)-$(RELEASE).src.rpm
diff --git a/lcovrc b/lcovrc
index 8669bdc..270fd95 100644
--- a/lcovrc
+++ b/lcovrc
@@ -1,5 +1,5 @@
 #
-# /etc/lcovrc - system-wide defaults for LCOV
+# /usr/share/defaults/lcov/lcovrc - system-wide defaults for LCOV
 #
 # To change settings for a single user, place a customized copy of this file
 # at location ~/.lcovrc
-- 
2.18.0

